### PR TITLE
feat: add remove_modal

### PR DESCRIPTION
**Pull request by MaskDuck**
## Summary

the reason to implement remove_modal is the same reason we should implement remove_view ig

## Checklist

- [ ] If code changes were made then they have been tested.
  - [ ] I have updated the documentation to reflect the changes.
  - [ ] I have run `task pyright` and fixed the relevant issues.
- [ ] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: nextcord/client.py [hunk 1]
```diff
@@ -1852,6 +1852,33 @@ def add_modal(self, modal: Modal, *, user_id: Optional[int] = None) -> None:
 
         self._connection.store_modal(modal, user_id)
 
+    def remove_modal(self, modal: Modal) -> None:
+        """Remove a :class:`nextcord.ui.Modal` from persistent listening.
+
+        Parameters
+        ------------
+        modal: :class:`nextcord.ui.Modal`
+            The view to register for dispatching.
```
Comment (MaskDuck) [inline nextcord/client.py:1861]:
```suggestion
            The view to remove from dispatching.
```

File: nextcord/client.py [hunk 2]
```diff
@@ -1852,6 +1852,33 @@ def add_modal(self, modal: Modal, *, user_id: Optional[int] = None) -> None:
 
         self._connection.store_modal(modal, user_id)
 
+    def remove_modal(self, modal: Modal) -> None:
+        """Remove a :class:`nextcord.ui.Modal` from persistent listening.
+
+        Parameters
+        ------------
+        modal: :class:`nextcord.ui.Modal`
+            The view to remove from dispatching.
```
Comment (DenverCoder1) [inline nextcord/client.py:1861]:
```suggestion
            The modal to remove from dispatching.
```

File: nextcord/client.py [hunk 3]
```diff
@@ -1852,6 +1852,33 @@ def add_modal(self, modal: Modal, *, user_id: Optional[int] = None) -> None:
 
         self._connection.store_modal(modal, user_id)
 
+    def remove_modal(self, modal: Modal) -> None:
+        """Remove a :class:`nextcord.ui.Modal` from persistent listening.
+
+        Parameters
+        ------------
```
Comment (ooliver1) [inline nextcord/client.py:1859]:
```suggestion
        Parameters
        ----------
```

File: nextcord/client.py [hunk 4]
```diff
@@ -1852,6 +1852,33 @@ def add_modal(self, modal: Modal, *, user_id: Optional[int] = None) -> None:
...
+        modal: :class:`nextcord.ui.Modal`
+            The modal to remove from dispatching.
+
+        Raises
+        -------
```
Comment (ooliver1) [inline nextcord/client.py:1864]:
```suggestion
        Raises
        ------
```

**Review by ooliver1 — CHANGES_REQUESTED** [review]

**Comment by koala9712** [comment]
1) Why is this still a draft
2) Please update original comment with appropriate checks in the checklist
3) Testing code?

**Comment by ooliver1** [comment]
https://github.com/nextcord/nextcord/pull/706#issuecomment-1200438016

**Pull request closed without merging**